LEGO1/lego/sources/misc/legostorage.h

```diff
@@ -156,4 +156,16 @@ class LegoFile : public LegoStorage {
 	FILE* m_file; // 0x08
 };
 
+template <class T>
+inline void ReadFromStorage(LegoStorage* p_storage, T* p_variable, LegoU32 p_size = sizeof(T))
```

Comment:
I think these are exactly the same as the two Read/Write templates that already exist further up in this file, or am I missing something?

Reply:
Well that explains the name collision I saw 🤦‍♂️ I probably won't have time to fix that before the weekend. On a related note, do you know if is there a concept in this version of C++ that would restrict access to `LegoStorage::Read` and `Write` without breaking anything, in order to force everyone to use the more high-level functions?

Reply:
You could make `Read` and `Write` private or protected and then only allow the high-level functions access to them using the `friend` keyword. However, there are many instances in the code base where the regular `Read`/`Write` functions do need to be publically accessed, so this isn't an option - in fact, as far as I can tell, the high level read/write functions are mainly used in the serializiation of the game's save data, but not anywhere else

Reply:
Thanks for the input! I'll have another look and see if something can be improved - will open another PR if I get somewhere. The code of this PR should finally be good now.